doc/developer/design/20220322_multi_table_transactions.md

```diff
@@ -0,0 +1,48 @@
+# Multi-table Transactions
+
+## Summary
+
+Upcoming persist changes will remove the persist API allowing writing to multiple collections in an atomic operation.
+In order to allow the persist API to be written, we must do one of:
+
+1. Disallow multi-table transactions.
+2. Use an existing (or upcoming) materialize API to record transactions.
+3. Use a third-party product to record transactions.
+
+We need a log to record transactions that can write the entire transaction atomically, which will allow the rest of the system to know that the transaction committed.
+The controller writes transactions to the WAL, then if committed, each collection can apply their portion of the writes asynchronously.
+There are two general methods to determine "if committed" are:
+- Writing to the log is controlled by a consensus operation such that any transaction successfully written to the log has committed.
+  Here the log needs a CAS operation during write.
+  [diagrams](https://materializeinc.slack.com/archives/C035TU9QF5W/p1647452686144389),
+  [description](https://materializeinc.slack.com/archives/C035TU9QF5W/p1647362440541409)
+
+- Readers of the log must use data contained in a log entry (and possibly previous entries) to determine whether a log entry committed.
```

Comment:
Ah, I think this line shows the crux of the divergence in our understanding. I think in this design, the readers determine if the log _can_ apply and the transaction commits iff they all agree; this relies on distributed deterministic execution _a la_ the Calvin protocol

 https://materializeinc.slack.com/archives/C035TU9QF5W/p1647454384556029